File: src/main/java/discord/commands/container/minion/Minion.java
package discord.commands.container.minion;

import discord.commands.DiscordCommand;
import discord.entity.DiscordUser;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;

public class Minion extends DiscordCommand {

    @Override
    public void execute(EmbedBuilder embed, StringBuilder text, MessageReceivedEvent bot, DiscordUser user, String... cmd) {

        if (user.getMinion() == null)
            user.setMinion(new boolean[] { false, false });

        if (!user.getMinion()[0]) {
            bot.getChannel().sendMessage("You do not have a minion! Purchase one today by doing -buyminion or -buym").queue();
            return;
        }

        bot.getChannel().sendMessage(user.getMinion()[1] ? "<:minion:718547445985968158>Minion_name is currently doing whatever... (TODO)" : "<:minion:718547445985968158>Minion_name is currently doing nothing. Commands:\n-setname <name>\nTODO MORE COMMANDS!").queue();

    }

    @Override
    public String[] getCommand() {
        return new String[] { "-minion", "-m" };
    }

    @Override
    public String getDescription() {
        return "Check what your minion is currently doing.";
    }

    @Override
    public String getArguments() {
        return null;
    }

}

File: src/main/java/discord/commands/container/kotlin/minion/Minion.kt
package discord.commands.container.kotlin.minion

import discord.commands.DiscordCommand
import discord.entity.DiscordUser
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class Minion : DiscordCommand() {

    override fun getArguments(): String? {
        return null
    }

    override fun getDescription(): String {
        return "Returns the status of your minion"
    }

    override fun getCommand(): Array<String> {
        return arrayOf("-minion", "-m")
    }

    override fun execute(embed: EmbedBuilder?, text: StringBuilder?, bot: MessageReceivedEvent?, user: DiscordUser, vararg cmd: String?) {

        if (!user.hasMinion) {
            bot?.channel?.sendMessage("You do not own a minion. Interested? Purchase one! -buyminion")?.queue()
            return
        }

        bot?.channel?.sendMessage("<:minion:718547445985968158> **${user.getMinionName()}** is currently doing nothing! Actions will be added soon.\n\n**Change your minions name!** (-setname <name>)")?.queue()

    }

}